addons/Cross_Lesson/src/presenter.js

```diff
@@ -210,10 +216,139 @@ function AddonCross_Lesson_create(){
+
+        if (upgradedModel['langAttribute'] === undefined) {
+            upgradedModel['langAttribute'] =  '';
+        }
+
+        return upgradedModel;
+    };
+
+    presenter.upgradeSpeechTexts = function AddonCross_Lesson_upgradeSpeechTexts (model) {
+        var upgradedModel = {};
+        jQuery.extend(true, upgradedModel, model); // Deep copy of model object
+
+        if (!model['speechTexts']) {
+            upgradedModel['speechTexts'] = {
+                GoToLesson: {GoToLesson: ""}
+            };
+        }
+
+        return upgradedModel;
+    };
+
     presenter.upgradeModel = function AddonCross_Lesson_upgradeModel (model) {
         var upgradedModel = presenter.upgradeOpenLessonInCurrentTab(model);
+        upgradedModel = presenter.upgradeLangTag(upgradedModel);
+        upgradedModel = presenter.upgradeSpeechTexts(upgradedModel);
         return upgradedModel;
     };
 
+    presenter.setSpeechTexts = function(speechTexts) {
+        presenter.speechTexts = {
+            GoToLesson: presenter.DEFAULT_TTS_PHRASES.goToLesson
+        };
+
+        if (!speechTexts || $.isEmptyObject(speechTexts)) {
+            return;
+        }
+
+        presenter.speechTexts = {
+            GoToLesson: presenter.getSpeechTextProperty(
+                speechTexts.GoToLesson.GoToLesson,
+                presenter.speechTexts.GoToLesson)
+        };
+    };
+
+    presenter.getSpeechTextProperty = function (rawValue, defaultValue) {
+        if (rawValue === undefined || rawValue === null) {
+            return defaultValue;
+        }
+
+        var value = rawValue.trim();
+        if (value === '') {
+            return defaultValue;
+        }
+
+        return value;
+    };
+
+    function CrossLessonKeyboardController (elements, columnsCount) {
+        KeyboardController.call(this, elements, columnsCount);
+    }
+
+    CrossLessonKeyboardController.prototype = Object.create(window.KeyboardController.prototype);
+    CrossLessonKeyboardController.prototype.constructor = CrossLessonKeyboardController;
+
+    presenter.buildKeyboardController = function () {
+        presenter.keyboardControllerObject = new CrossLessonKeyboardController(presenter.getElementsForKeyboardNavigation(), 1);
+    };
+
+    presenter.getElementsForKeyboardNavigation = function () {
+        return this.$view.find(".cross-lesson-wrapper");
+    };
+
+    presenter.keyboardController = function(keycode, isShiftKeyDown, event) {
+        presenter.keyboardControllerObject.handle(keycode, isShiftKeyDown, event)
+    };
+
+    CrossLessonKeyboardController.prototype.selectAction = function () {
+        this.getTarget(this.keyboardNavigationCurrentElement, true)[0].click();
+    };
+
+    CrossLessonKeyboardController.prototype.getTarget = function (element, willBeClicked) {
+        return $(element);
+    };
+
+    CrossLessonKeyboardController.prototype.mark = function (element) {
+        return;
```

Comment:
brak ciała funkcji - czy ta metoda nie powinna dodawać klasy css odpowiedzialnej za zaznaczenie obiektu?

Reply:
Tutaj specjalnie w ten sposób robię, żeby nadpisać domyślną funkcję. Wewnątrz cross lesson jest tylko jeden element. Przy zaznaczeniu go dodatkowo tą funkcją wygląda to dziwnie (dwa style nałożone na ten sam element).